### ISSUE TITLE
Question about training experiments on modelnet40

### ISSUE THREAD

**Issue by CorgiBoyG** [yaorz97/PARENet#2]
Thank you for the work you have done, which achieves higher performance on the point cloud registration benchmarks. My research interest is in partial-overlap 3D registration. Recently I have tried to train on modelnet40, but I found that I often get errors in the middle of my run. Could you please provide some suggestions to help me in order to solve this problem. Thanks again for doing such a superior job and I look forward to hearing from you.

**Comment by yaorz97** [comment]
Thanks for your attention! Could you provide more details about the errors you encountered?

**Comment by CorgiBoyG** [comment]
> Thanks for your attention! Could you provide more details about the errors you encountered?

![image](https://github.com/user-attachments/assets/f3831b4c-e4a4-492c-b1c3-84e5b9aa55e2)

**Comment by yaorz97** [comment]
It looks like the number of selected points $k$ is greater than the number of all points.  Are point clouds being oversampled, causing them to be too sparse? Using a smaller voxel size for downsampling or reducing the value of $k$ may address this issue.

**Comment by CorgiBoyG** [comment]
> It looks like the number of selected points k is greater than the number of all points. Are point clouds being oversampled, causing them to be too sparse? Using a smaller voxel size for downsampling or reducing the value of k may address this issue.

Yes, I set the parameters of PARE-Net for modelnet40 training based on PARE-Net's training parameters on 3dmatch and GeoTransformer's training parameters on modelnet40. Where the initial voxel size is 0.025 and its the same as 3dmatch. Now I adjusted it even smaller and it doesn't return errors anymore. But I have encountered a new problem where the calculated fine_re_loss is nan. on line 93 of _loss.py_ “ **batch_indices, ref_indices, src_indices = torch.nonzero(neg_map, as_tuple= True)**” the returned **ref_indices** and **src_indices** are often **empty**. Thanks again for your reply.
![image](https://github.com/user-attachments/assets/b63d8efe-1c3a-464e-9453-c914c3a69475)
![image](https://github.com/user-attachments/assets/3bbbf1c8-ae6f-43e5-9538-43313b127258)
![image](https://github.com/user-attachments/assets/c2c7dcd4-f515-492a-a886-ef29aeeb3ace)

**Comment by yaorz97** [comment]
How do you set the hyper-parameter `self.negative_radius`, you could use a smaller one to address this issue. Note that `self.negative_radius`  should always be greater than `self.positive_radius`.

**Comment by CorgiBoyG** [comment]
> How do you set the hyper-parameter `self.negative_radius`, you could use a smaller one to address this issue. Note that `self.negative_radius` should always be greater than `self.positive_radius`.

I am very sorry for not replying to your comments in time. I tried before to set `self.negative_radius` and `self.positive_radius` to **0.075** and **0.05** respectively. The training of the network is fine, but during the evaluation of the inference there will be an error of **_f_ri_loss being nan_**, which leads to no decrease in error metrics in the validation set, for example the numerical result of **_RRE is 71_**, which is not possible. May I ask if you have experimented on the modelnet40 dataset before and got some good results? Looking forward to your reply again, it's important for me, thanks.
![image](https://github.com/user-attachments/assets/f90dab4b-b193-475e-be8f-d1a5ae02cc6c)
![image](https://github.com/user-attachments/assets/7a0b0c96-e566-40df-a57f-512ed22c27cc)

**Comment by yaorz97** [comment]
We have not experimented on ModelNet40, but the over-sparse superpoints seem to cause this error. We found that GeoTrans uses a different backbone on ModelNet40, and you can reduce a downsample layer like GeoTrans to avoid this issue.

**Comment by CorgiBoyG** [comment]
> We have not experimented on ModelNet40, but the over-sparse superpoints seem to cause this error. We found that GeoTrans uses a different backbone on ModelNet40, and you can reduce a downsample layer like GeoTrans to avoid this issue.

I have tried to modify it, but it always reports an error. I removed the encoder_4, but I'm confused about the code in the forward reasoning, can you modify it and provide a copy of the code for the backbone, thank you!

**Comment by jun3333333** [comment]
@CorgiBoyG Hello, I am very interested in the work you have done. I am also encountering difficulties at the same step as you did and want to apply this code to a small-scale dataset like ModelNet40. However, I have run into the same issue as yours during the implementation process. Please feel free to contact me via QQ: 2373178085@qq.com.

**Comment by quannever** [comment]
@CorgiBoyG Hi,  if the issue with f_ri_loss becoming NaN has been resolved? If so, could you please share how it was addressed?
